added movies list

apostolclaudiaSV opened the pull request:
added movie list and displayed it in a table view according to the selected tab

apostolclaudiaSV committed:
added mockup movie list and displayed it on screen according to selected tab

apostolclaudiaSV committed:
fixed color for tab bar and navigation bar when scrolling

apostolclaudiaSV committed:
changed nav bar background color to clear to suit light and dark mode

apostolclaudiaSV committed:
implemented FavoritesVC that inherits BaseVC
added  sortAndFilter method in order to simultaneously sort and filter the movies array
moved each struct into its own separate file

ancaCalugar commented on MoviesApp/Model/MoviesListProvider.swift:
```diff
@@ -0,0 +1,82 @@
+        return filteredMovies
+    }
+    
+    func sortAndFilter(sortCriteria: SortCriteria, filterCriteria: FilterCriteria) -> [Movie] {
```
you should use here to an input of movies (maybe for now have unsortedMovies be a static var and provide it as default value for the movies parameter)..

ancaCalugar commented on MoviesApp/Controller/BaseViewController.swift:
```diff
@@ -0,0 +1,50 @@
+
+    let movieList = MoviesListProvider()
+    var filteredMovies: [Movie] = []
+    var sortCriteria: SortCriteria = .popularityDesc
```
how about having the criteria properties be read-only (and overridable) so that in Favorites you don't have to override viewDidLoad, only the two properties?

apostolclaudiaSV committed:
removed unused prototype cell
made sortCriteria and filterCriteria read-only
added input of movies to sortAndFilter function

apostolclaudiaSV committed:
dequeued cell
